code/ai/aigoals.cpp

```diff
@@ -1245,115 +1247,145 @@ int ai_remove_goal_sexp_sub( int sexp, ai_goal* aigp )
 	/* The operator to use */
 	int op = get_operator_const( node );
 
+	// since this logic is common to all goals removed by the remove-goal sexp
+	auto eval_priority_et_seq = [sexp, &remove_more](int n, int priority_if_no_n = -1)->int
```

Comment:
It's an abbreviated "and the following nodes", i.e. evaluate the priority and the nodes that come after the goal operator: the remove-all flag and the check-priority flag.  I imagine that a future enhancement would add another boolean flag for checking the goal target, since right now `remove-goal` does not distinguish between the same goal with different targets.